bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java

```diff
@@ -328,6 +316,7 @@ private void run() {
 
                 item = group.get();
 
+                Strand previousStrand = numThreads == 1 ?  strandHolder.get().strand : null;
```

Comment:
Shall we move this out of while loop. Do we need if check here.
Strand previousStrand = strandHolder.get().strand

Reply:
Moved to a class variable and assigned in constructor with 5aa13c7